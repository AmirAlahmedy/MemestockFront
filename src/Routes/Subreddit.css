.subredditFixed{
    width:100%;
    position:fixed;
    top:49px; 
}

.subredditContainer{
    width:80%;
    margin:auto;
    overflow: hidden;
}
.subredditName {
    font-size: 22px;
    font-weight: 500;
    color: #fcfcfb;
}
#showcase{
    top:49;
    background-image: url('../images/Covertest.jpg');
    background-position: initial;
    background-color:grey;
    background-size: 100%;
    background-repeat: none;
    min-height: 150px;
    text-align: left;
    margin-bottom:20px;
    cursor:pointer;
}
#showcase h1{
    font-style: inherit;
    font-size:22px;
    font-weight: 500;
    float:left;
    margin:0;
    padding-top:50px;
    padding-left:1%;
    border:0;
}
#showcase img{
    margin-top:40px;
    margin-left:3%;
    float:left;
    border-radius:1000px;
    block-size: 50px;
}
@media(max-width:600px){
    #main{
        width:100%;
        float:none;
    }
    #sidebar{
        width:100%;
        float:none;
    }
}
@media(max-width:1000px){
    #showcase{
        background-size: 100% 6000px;
    }

}