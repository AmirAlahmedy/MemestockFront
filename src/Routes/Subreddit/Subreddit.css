.subredditFixed{
    width:100%;
    top:49px; 
}

.subredditContainer{
    width:80%;
    margin:auto;
    overflow: hidden;
}
.subredditName {
    font-size: 22px;
    font-weight: 500;
    color: #fcfcfb;
}
#subredditShowcase{
    top:49;
    background-image: url('../../assets/images/Covertest.jpg');
    background-position: initial;
    background-color:grey;
    background-size: 100%;
    background-repeat: none;
    min-height: 250px;
    text-align: left;
    cursor:pointer;
}
#subredditShowcase h1{
    font-style: inherit;
    font-size:22px;
    font-weight: 500;
    float:left;
    margin:0;
    padding-top:110px;
    padding-left:1%;
    border:0;
}
#subredditShowcase img{
    margin-top:100px;
    margin-left:3%;
    float:left;
    border-radius:1000px;
    block-size: 50px;
}

#subredditPageContent{
    float:left;
    width:68%;
    box-sizing:border-box;
    margin-top:50px;
    margin-bottom:100px;
    margin-right:10px;
    
}
.subredditPageThread { 
    margin-bottom:20px; 
    /*
    background-color: rgb(26, 26, 27);
    
    border-bottom: 1px solid rgb(52, 53, 54);
    border-right:  1px solid rgb(52, 53, 54);
    border-left:  1px solid rgb(52, 53, 54);
    border-top:  1px solid rgb(52, 53, 54);
    border-radius: 6px; 
    */
}
#subredditSidebarContainer{
    width:30%;
    float:right;
    padding:0;
    box-sizing:border-box;
    margin-top:50px;
}
.subredditSidebarComponent{
    margin-bottom:15px;
    background-color: rgb(26, 26, 27);
    border-bottom: 1px solid rgb(52, 53, 54);
    border-right:  1px solid rgb(52, 53, 54);
    border-left:  1px solid rgb(52, 53, 54);
    border-top:  1px solid rgb(52, 53, 54);
    border-radius: 6px;
}
#subredditNavbar{
    background-color:black;
    /*
    border-radius:5px;
    background-color:rgb(26, 26, 27);
    
    border: 1px solid rgb(52, 53, 54);
    */
    color:white;
}
#subredditSort{
    /*border-radius:5px; */
    background-color:rgb(26, 26, 27);    
    border-top: 1px solid rgb(52, 53, 54);
    border-bottom: 1px solid rgb(52, 53, 54);
    color:white;
}
.srLinks{
    float:left;
    margin:0;
    padding-top:0;
    padding-right:0;
    padding-left:0;
    padding-bottom:7px;
    list-style: none;
}
.srLinks a{
    color:white;
    text-decoration: none;
    font-size:18px;
    line-height: 1em;
    padding:10px;
    text-align:center;
}
.srLinks a:hover{
    color: rgb(233, 172, 92);
}

.srSortdropdownMenu{
    padding-bottom: 10px;
    float:left;    
}
.srSortdropButton{
    background: transparent;
    cursor: pointer;
    border: none;
    color: #f2f2f2;
    font-size: 12px;
    line-break: auto;
}
.srSortdropUl{
    font-size:20px;
    display: flex;
    flex-direction: column;
    list-style-type: none;
    padding:0;
}
.srdropList{
    background-color: rgb(26, 26, 27);
    display: none;
    position: absolute;
    min-width: 160px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
    padding: 10px;
    z-index: 1;
    border-bottom: 1px solid rgb(52, 53, 54);
    border-right:  1px solid rgb(52, 53, 54);
    border-left:  1px solid rgb(52, 53, 54);
    border-radius: 6px;
}
.srSortdropdownMenu:hover .srdropList{
    display: block;
}
.srSortdropdownItem{
    border:1px solid rgb(52, 53, 54);
    float: 1;
    list-style: none;
}
.srsort{
    text-decoration: none;
    float: left;
    color:rgb(111, 112, 113);
}
.srsort:hover{
    color:rgb(233, 172, 92);
}
.subredditSidebarComponent {
    font-size:12px;
    text-align: left;
    padding-left:20px;
    padding-right:20px;
    padding-bottom:15px;
    color:rgb(132, 134, 136);
}
.srSidebarCompTitle {
    font-size:24px;
    font-weight: 500;
    text-align: left;
    color: lightgrey;
    padding-top: 10px;
    padding-bottom: 20px;
}
.srSidebarCompTitle img{
    margin-left:0px;
    margin-right:10px;
    float:left;
    border-radius:1000px;
    block-size: 30px;
}
.srSidebarSubscribers {
    margin-left:10px;
    color:lightgrey;
    font-size:26px;
    font-weight: 700;
    margin-bottom:10px;
}
.srSidebarBio { 
    margin-left:5px;
    padding-right:10px;
    color:lightgrey;
    font-size:16px;
}
.srSidebarSubscribeButton {
    margin:10px 0px;
    padding:10px;
    width:100%;
    border-radius: 7px;
    background-color:lightgrey;
    font-size:18px;
    font-weight: 600;
    cursor:pointer;
}
.srSidebarSubscribeButton:hover{
    color:rgb(233, 172, 92);
    background-color:rgb(26, 26, 27);
}
.moderatorSr {
    cursor:pointer;
    font-size:20px;
    padding-bottom:5px;
    margin:0;
    color:rgb(116, 193, 219);
}
.rulesSr{
    cursor:pointer;
    font-size:20px;
    padding-bottom:5px;
    margin:0;
    color: #fcfcfb;
}

.moderatorSr:hover{
    text-decoration: underline;
}
#subreddit-footer{
    padding-top:30px;
    margin-top:40%;
    font-size:12px;
   margin-bottom:100px;
}
@media(min-width:900px){
    #subredditShowcase img{
    margin-left:200px;   
 }

    /*
    #main{
        width:100%;
        float:none;
    }
    
    #sidebar{
        width:100%;
        float:none;
    }
    */
}
/*
@media(max-width:1000px){
    #showcase{
        background-size: 200%;
    }

}
*/