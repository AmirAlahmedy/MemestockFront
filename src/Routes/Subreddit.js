import React, { Component } from 'react';
import './Subreddit.css';
import defImage from '../images/subreddit.png';
import Listing from '../Components/Listings';
import Thread from './Thread';

export class Subreddit extends Component {
    state ={
        name:'Jazztheory',
        threads:[]
    }
  render() {
    return (
      <div className="subredditFixed">
        <section id="subredditShowcase">
            <img src={defImage} alt="Subreddit Default"/>
            <h1>r/{this.state.name}</h1>
        </section>
        <div class="subredditContainer">
            <section id="subredditPageThreads">
                {/*<Listing/>*/}
                <Thread/>
            </section>
            <aside id="subredditSidebar">
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum similique, laudantium omnis, nostrum impedit non, itaque placeat expedita inventore numquam nam velit dolorem in consequuntur eaque quis officiis necessitatibus harum.
            </aside>
        </div>
      </div>
    )
  }
}

export default Subreddit
