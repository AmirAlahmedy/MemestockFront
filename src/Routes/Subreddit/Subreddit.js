import React, { Component } from 'react';
import './Subreddit.css';
import defImage from '../../assets/images/subreddit.png';
import Listings from '../../Components/Listings/Listings';
import Thread from '../Thread/Thread';
import { Link, Route, Switch, Router } from 'react-router-dom';
import data from '../../Mocks/subreddit-data.json';

import axios from '../../axios-orders';


let inDev = false;

export class Subreddit extends Component {
  state = {
    name: window.location.href.split("/").pop(),
    bio: '',
    threads: [],
    moderators: '',
    moderatorsList: [],
    subscribers: 0,
    subscribersList: [],
    date: '',
    rules: [],
    posts: [],
    subscribed: false,
    adminview: false,
    threadCreation: false,
    subredditEdit: false,
    threadsContent: [],
    flair: null,
    spoiler: false,
  }

  componentDidMount() {

    console.log("mounted");

    if (inDev === false) // && ginprodReducer.globalInProduction)
    {
      let srName = this.state.name;
      axios.get('/sr/' + srName + '/meta')
        .then(resp => {
          console.log(resp);
          console.log(resp.data);
          if (resp.status == 200) {
            console.log(resp.data);
            this.setState({
              name: this.state.name,
              bio: resp.data.Bio,
              threads: resp.data.posts,
              moderators: resp.data.adminUsername,
              moderatorsList: resp.data.modUsername ? resp.data.modUsername : [resp.data.adminUsername],
              subscribers: resp.data.subscribed_users.length,
              subscribersList: resp.data.subscribed_users,
              subscribed: resp.data.subscribed_users.includes(localStorage.getItem("Username")) ? true : false,
              adminview: resp.data.modUsername.includes(localStorage.getItem("Username")) ? true : false,
              date: resp.data.date,
              rules: resp.data.rules,
              cover: resp.data.subredditFile
            }, () => {
              const isAdmin = this.state.moderators === localStorage.getItem("Username")
                || this.state.moderatorsList.includes(localStorage.getItem("Username"));
              this.setState({
                adminview: isAdmin,
                //subscribed: isAdmin
              })
            })
            for (const threadID of resp.data.posts) {
              axios.get(`/sr/${srName}/thread/${threadID}`)
                .then(resp => {
                  if (resp.data && resp.status === 200) {
                    let threads = this.state.threadsContent;
                    threads.push(resp.data);
                    this.setState({
                      threadsContent: threads
                    });
                  }
                });
            }
          }
          else if (resp.status === 404) {
            alert("Subreddit Not Found");
            return Response.json;
          }
        })
        .catch(error => {
          // alert(cons);
          console.log(error);
        })
    }
    else {

      this.setState({
        bio: data.subreddit.bio,
        threads: data.subreddit.threads,
        moderators: data.subreddit.adminUsername,
        subscribers: data.subreddit.subscribers,
        date: data.subreddit.date,
        rules: data.subreddit.rules,
      });
    }
    let token = localStorage.getItem("token");
    let username = localStorage.getItem("Username");
    for (let i = 0; i < this.state.subscribers; i++) {
      if (username == this.state.subscribersList[i] || username == this.state.moderatorsList[i] || username == this.state.moderatorsList) {
        this.setState({
          subscribed: true
        })
      }
      if (username == this.state.moderators) {
        this.setState({
          adminview: true
        })
      }
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  };



  handleChange(e) {
    this.setState({
      spoiler: e.target.checked
    });
  }

  /**
   * For sending a subscribe request to the backend and updating the subscribed boolean state
   * @function srSubscribe
   * @param {event} - onClick event 
   */
  srSubscribe = (e) => {
    e.preventDefault();
    console.log('Subscribe Clicked');
    let headers = {
      auth: localStorage.getItem("token")
    }
    let SubredditName = this.state.name;
    axios.post('/sr/' + SubredditName + '/subs', null, { "headers": headers })
      .then(res => {
        if (res.status == 200) {
          console.log(res);

          console.log('subscribed!')
          this.setState({
            subscribed: true
          }
          );
        } else if (res.status === 404) {
          alert("Subreddit Not Found");
          return Response.json;
        }
      })
      .catch(error => {
        alert("Error Caught");
      })
  }
  /**
    * For sending an Unsubscribe request to the backend and updating the subscribed boolean state
    * @function srUnSubscribe
    * @param {event} - onClick event 
    */
  srUnSubscribe = (e) => {
    e.preventDefault();
    console.log('Unsubscribe Clicked');
    let headers = {
      auth: localStorage.getItem("token")
    }
    let SubredditName = this.state.name;
    axios.delete('/sr/' + SubredditName + '/subs', { "headers": headers })
      .then(res => {
        if (res.status == 200) {
          console.log(res);
          this.setState({
            subscribed: false
          }
          );
        } else if (res.status === 404) {
          alert("Subreddit Not Found");
          return Response.json;
        }
      })
      .catch(error => {
        alert("Error Caught");
      })
  }
  /**
    * For sending an delete request to the backend to delete the entire subreddit from the database
    * @function srUnSubscribe
    * @param {event} - onClick event 
    */
  delSubreddit = (e) => {
    e.preventDefault();
    console.log('Del Subreddit Clicked');
    let headers = {
      auth: localStorage.getItem("token")
    }
    let SubredditName = this.state.name;
    axios.delete('/sr/' + SubredditName, { "headers": headers })
      .then(res => {
        console.log(res);
        if (res.status == 200) {
          alert("Subreddit Deleted Successfully!");
          this.setState({
            subscribed: false
          }
          );
        }
        else if (res.status === 401) {
          alert("You're Not Authorised");
          return Response.json;
        }
      })
      .catch(error => {
        alert("Error Caught");
      })

  }
  /**
    * For changing the GUI and showing the fields for the thread creation
    * @function createThreadSidebar
    * @param {event} - onClick event 
    */
  createThreadSidebar = (e) => {
    e.preventDefault();
    console.log('Clicked on create thread sidebar');
    if (this.state.subscribed == false) {
      //CANNOT CREATE A POST UNLESS SUBSCRIBED
      alert('CANNOT CREATE A POST UNLESS SUBSCRIBED');
    }
    else {
      this.setState({
        threadCreation: true
      })

    }
  }
  /**
         * For changing the GUI and hiding the fields for the thread creation
         * @function CancelCreation
         * @param {event} - onClick event 
         */
  CancelCreation = (e) => {
    e.preventDefault();
    console.log('Clicked on Cancel thread sidebar');
    this.setState({
      threadCreation: false
    })
  }
  editSubreddit = (e) => {
    e.preventDefault();
    console.log("Clicked on the edit subredditbutton");
    if (this.state.adminview == false) {
      //CANNOT CREATE A POST UNLESS SUBSCRIBED
      alert('CANNOT EDIT A SUBREDDIT UNLESS IS MODERATOR');
    }
    else {
      this.setState({
        subredditEdit: true
      })
    }
  }
  cancelSubreddit = (e) => {
    e.preventDefault();
    console.log('Clicked on Cancel thread sidebar');
    this.setState({
      subredditEdit: false
    })
  }

  /**
   * For sending an post request to the backend and creating a thread in this subreddit
   * @function handleSubmit
   * @param {event} - onClick event 
   */
  handleSubmit = (e) => {
    e.preventDefault();

    const srdata = {
      "title": document.getElementById("threadTitleField").value,
      "threadBody": document.getElementById("threadBodyField").value,
      "spoiler": this.state.spoiler
    }
    if (this.state.imagePost) {
      srdata.base64image = this.state.imagePost
    }
    let headers = {
      auth: localStorage.getItem("token")
    }
    let SubredditName = this.state.name;
    axios.post('/sr/' + SubredditName + '/thread', srdata, { "headers": headers })
      .then(res => {
        console.log(res);
        if (res.status == 200) {
          alert("Thread Created Successfully!");
        } else if (res.status === 401 || res.status === 404) {
          alert("Thread Creation Unsuccessful");
          return Response.json;
        }
      })
      .catch(error => {
        alert("Errorrrrrrr Caught");
      })
  }

  handleEdit = (e) => {
    e.preventDefault();
    let newmods = this.state.moderatorsList;
    const newMod = document.getElementById("subredditmoderatorField").value;
    if (newMod) {
      newmods.push(newMod);
    }
    const srdata = {
      "newName": document.getElementById("subredditNameField").value,
      "newRules": [document.getElementById("subredditRule1Field").value, document.getElementById("subredditRule2Field").value, document.getElementById("subredditRule3Field").value],
      "newBio": document.getElementById("subredditBioField").value,
      "newMods": newmods,
      "base64image": this.state.image
    }

    let headers = {
      auth: localStorage.getItem("token")
    }
    let SubredditName = this.state.name;
    axios.put('/sr/' + SubredditName, srdata, { "headers": headers })
      .then(res => {
        console.log(res);
        if (res.status == 200) {
          alert("Thread Created Successfully!");
          let srName = this.state.name;
          axios.get('/sr/' + srName + '/meta')
            .then(resp => {
              console.log(resp);
              if (resp.status == 200) {
                console.log(resp.data.rules);
                console.log("length", resp.data.posts.length)
                window.location.href = "/r/" + SubredditName;
              }
            })
        } else if (res.status === 401 || res.status === 404) {
          alert("Thread Creation Unsuccessful");
          return Response.json;
        }
      })
      .catch(error => {
        alert("Error Caught");
      })
  }
  getThreads() {
    if (!this.state.threadsContent) return;
    return this.state.threadsContent.map(thr =>
      <Thread
        id={thr._id}
        username={thr.creatorUsername}
        subreddit={thr.subredditName}
        title={thr.title}
        content={thr.body}
        upvotes={thr.votes}
      />);
  }
  /**
  * For changing the GUI and hiding the fields for the thread creation
  * @function CreateFlair 
  * @param {event} - Submition event 
  */
  CreateFlair(e) {
    e.preventDefault();

    let body = {
      srName: this.state.name,
      flair: e.target.querySelector('.flairInput').value
    }
    let headers = {
      'auth': localStorage.getItem('token')
    }
    axios.post('/user/CreateFlair', body, { "headers": headers })
      .then(response => {
        console.log('flair response', response);
        //Na hash3'al l server bta3 l backend 3andi 3shan hazbot feeh 7agat 
        //Ha sharo fa l mafrod ysht3'al...
        this.setState({
          flair: body.flair
        })
      })
      .catch(error => {

      })
  }
  handleNewImage(e) {
    const files = e.target.files;
    if (!files || !files.length) return;

    const reader = new FileReader();

    reader.onload = (eReader) => {
      if (!eReader.target || !eReader.target.result) return;
      this.setState({
        image: eReader.target.result
      });
    }

    reader.readAsDataURL(files[0]);

  }
  handleNewImagePost(e) {
    const files = e.target.files;
    if (!files || !files.length) return;

    const reader = new FileReader();

    reader.onload = (eReader) => {
      if (!eReader.target || !eReader.target.result) return;
      this.setState({
        imagePost: eReader.target.result
      });
    }

    reader.readAsDataURL(files[0]);

  }
  render() {
    console.log(this.state.flair);
    return (
      <div className="subredditFixed">
        <section id="subredditShowcase" style={{ backgroundImage: this.state.cover ? `url(http://18.217.163.16/api${this.state.cover})` : null }}>
          <img src={defImage} alt="Subreddit Default" />
          <h1>r/{this.state.name}</h1>
        </section>

        {/*<nav id="subredditNavbar">
          <div className="subredditContainer">
            <div id="subredditSort">
              <div className="subredditContainer">

                <div className="srSortdropdownMenu">
                  <button className='srSortdropButton'>
                    Sorting <i class="downIcon fas fa-sort-down"></i>
                  </button>
                  <div role='menu' className='srdropList'>
                    <ul className='srSortdropUl'>
                      <li className='srSortdropdownItem' className='sort toHome'>Hot</li>
                      <li className='srSortdropdownItem' className='sort toHome'>New</li>
                      <li className='srSortdropdownItem' className='sort toHome'>Controversial</li>
                      <li className='srSortdropdownItem' className='sort toHome'>Top</li>
                      <li className='srSortdropdownItem' className='sort toHome'>Rating</li>
                    </ul>

                  </div>
                </div>
              </div>
            </div>

          </div>
        </nav>*/}
        <div class="subredditContainer">
          <section id="subredditPageContent">
            <div>
              {this.getThreads()}

            </div>
          </section>

          <aside id="subredditSidebarContainer">
            <div className="subredditSidebarComponent">
              <h5>COMMUNITY DETAILS</h5>
              <div className="srSidebarCompTitle">
                <img src={defImage} alt="Subreddit Default" />
                r/{this.state.name}
              </div>
              <div className="srSidebarSubscribers">
                {this.state.subscribers} <br />
                Subscribers
                </div>
              <div className="srSidebarBio">
                <p>{this.state.bio}</p>
              </div>
              {
                this.state.moderators === localStorage.getItem("Username") ? <span></span>
                  : this.state.subscribed ? <button className="srSidebarSubscribeButton" onClick={this.srUnSubscribe}>UNSUBSCRIBE</button>
                    : <button className="srSidebarSubscribeButton" onClick={this.srSubscribe}>SUBSCRIBE</button>
              }
              <button className="srSidebarSubscribeButton" onClick={this.createThreadSidebar}>CREATE A POST</button>
            </div>
            {
              this.state.subredditEdit ?
                <div className="subredditSidebarComponent">
                  <h5>EDIT SUBREDDIT</h5>
                  <hr></hr>
                  <form onSubmit={this.handleEdit}>
                    <div className="formGroupSrComponent">
                      <label for="Subreddit New Name">Enter Subreddit Name</label>
                      <textarea type="textarea" name="text" id="subredditNameField" placeholder="Enter Name Here" />
                    </div>
                    <div className="formGroupSrComponent">
                      <label for="Rule1">Enter Rule </label>
                      <textarea type="textarea" name="text" id="subredditRule1Field" placeholder="Enter Rule Here" />
                    </div>
                    <div className="formGroupSrComponent">
                      <label for="Rule2">Enter Rule </label>
                      <textarea type="textarea" name="text" id="subredditRule2Field" placeholder="Enter Rule Here" />
                    </div>
                    <div className="formGroupSrComponent">
                      <label for="Rule3">Enter Rule </label>
                      <textarea type="textarea" name="text" id="subredditRule3Field" placeholder="Enter Rule Here" />
                    </div>
                    <div className="formGroupSrComponent">
                      <label for="Bio">Enter Bio </label>
                      <textarea type="textarea" name="text" id="subredditBioField" placeholder="Enter Bio Here" />
                    </div>
                    <div className="formGroupSrComponent">
                      <label for="Bio">Add a new moderator </label>
                      <textarea type="textarea" name="text" id="subredditmoderatorField" placeholder="Enter Moderator Here" />
                    </div>
                    <div className="formGroupSrComponent">
                      <label for="cover">Cover </label>
                      <input type="file" name="cover" id="cover" onChange={this.handleNewImage.bind(this)} />
                    </div>
                    <button className="srSidebarSubscribeButton">Edit Subreddit</button>
                    <button type="button" className="srSidebarSubscribeButton" onClick={this.cancelSubreddit}>CANCEL</button>
                  </form>
                </div> : <div></div>
            }
            {
              this.state.threadCreation ?
                <div className="subredditSidebarComponent">
                  <h5>CREATE A POST</h5>
                  <hr></hr>
                  <form onSubmit={this.handleSubmit}>
                    <div className="formGroupSrComponent">
                      <label for="ThreadTitle">Enter Title</label>
                      <textarea type="textarea" name="text" id="threadTitleField" placeholder="Enter Title Here" />
                    </div>
                    <div className="formGroupSrComponent">
                      <label for="ThreadBody">Enter Thread Body</label>
                      <textarea type="textarea" name="text" id="threadBodyField" placeholder="Enter Body Here" />
                    </div>
                    <div className="formGroupSrComponent">
                      <label for="coverPost">Cover </label>
                      <input type="file" name="coverPost" id="coverPost" onChange={this.handleNewImagePost.bind(this)} />
                    </div>
                    <div className="formGroupThreadCheckbox">
                      <label className="spoilerLabel" for="Spoiler">Is it a spoiler?</label>
                      <input type="checkbox" name="Spoiler" id="Spoiler" onChange={this.handleChange} value={this.state.spoiler} />
                    </div>
                    <button className="srSidebarSubscribeButton" >CREATE</button>
                    <button className="srSidebarSubscribeButton" onClick={this.CancelCreation}>CANCEL</button>
                  </form>
                </div> : <div></div>
            }
            <div className="subredditSidebarComponent">
              <h5>MODERATORS & ADMINS</h5>
              <ul>
                {/*<li className="moderatorSr" ><Link to={`/user/${this.state.moderators}`}>u/{this.state.moderators}</Link></li> */}
                {
                  this.state.moderatorsList.map(moderator => {
                    return (
                      <li className="moderatorSr" key={moderator}><Link to={`/user/${this.state.moderators}`}>u/{this.state.moderators}</Link></li>
                    );
                  })
                }
              </ul>
            </div>
            <div className="subredditSidebarComponent">
              <h5>RULES</h5>
              <ol>
                {
                  this.state.rules.map(rule => {
                    return (
                      <li className="rulesSr" key={rule}>{rule}</li>
                    );
                  })
                }
              </ol>
            </div>

            {
              this.state.adminview ? <button className="srSidebarSubscribeButton" onClick={this.editSubreddit}>Edit Subreddit</button> : <div></div>
            }
            {
              this.state.adminview ? <button className="srDeleteButton" onClick={this.delSubreddit}>Delete Subreddit</button> : <div></div>
            }
            <div className="subredditSidebarComponent flairComp">
              <h5>CREATE YOUR OWN NAME</h5>
              <form id='flairForm' onSubmit={this.CreateFlair.bind(this)}>
                <input
                  className='flairInput'
                  type='text'
                  placeholder='Flair' />
                <button className="srSidebarSubscribeButton" type='submit' >Create a flair</button>
              </form>

            </div>

          </aside>

        </div>
      </div >
    )
  }
}

export default Subreddit
