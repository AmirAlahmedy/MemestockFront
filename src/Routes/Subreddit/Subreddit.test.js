import React from "react";
import { create } from "react-test-renderer";
import Subreddit from "./subreddit";

describe("Subreddit component snapshot", () => {
  test("it matches the snapshot", () => {
    const component = create(<Subreddit />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

it("Shows subreddit page meta", async () => {
  const component = create(<Subreddit />);
  const instance = component.getInstance();
  await instance.componentDidMount();
  console.log(instance.state); 
});

test("Buttons", () => {
  const component = create(<Subreddit  />);
  const rootInstance = component.root;
  const button = rootInstance.findByType("button");
  button.props.onClick();
  expect(button.props.children).toHaveBeenCalled();
});
//subscribed={false}
/*
describe("Subscribe Button", () => {
  test("it shows the expected text when clicked", () => {
    const component = create(<Subreddit />);
    const instance = component.getInstance();
    //expect(instance.state.text).toBe("");
    instance.editSubreddit();
    expect(instance.state.subredditEdit).toBe(true);
  });
});
*/
/*
it('Login should call handleSubmit when clicked', () => {
  const spy = jest.spyOn(Subreddit.prototype, 'handleChange');
  const component = create(<Subreddit />);
    const instance = component.getInstance();
  //const f_wrapper = create(<Subreddit/>);
  instance.find('.logInForm').simulate('submit');
  expect(spy).toHaveBeenCalled() 
});
/*
test("it shows the expected text when clicked", () => {
  const component = create(<Subreddit  />);
  const rootInstance = component.root;
  const button = rootInstance.findByType("button");
  button.props.onClick();
  expect(button.props.children).toBe("SUBSCRIBE");
});
*/
