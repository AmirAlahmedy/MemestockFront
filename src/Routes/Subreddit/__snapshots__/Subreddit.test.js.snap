// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature component it matches the snapshot 1`] = `
<div
  className="subredditFixed"
>
  <section
    id="subredditShowcase"
    style={
      Object {
        "backgroundImage": null,
      }
    }
  >
    <img
      alt="Subreddit Default"
      src="subreddit.png"
    />
    <h1>
      r/
      
    </h1>
  </section>
  <nav
    id="subredditNavbar"
  >
    <div
      className="subredditContainer"
    >
      <div
        id="subredditSort"
      >
        <div
          className="subredditContainer"
        >
          <div />
        </div>
      </div>
    </div>
  </nav>
  <div
    class="subredditContainer"
  >
    <section
      id="subredditPageContent"
    >
      <div />
    </section>
    <aside
      id="subredditSidebarContainer"
    >
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          COMMUNITY DETAILS
        </h5>
        <div
          className="srSidebarCompTitle"
        >
          <img
            alt="Subreddit Default"
            src="subreddit.png"
          />
          r/
          
        </div>
        <div
          className="srSidebarSubscribers"
        >
          0
           
          <br />
          Subscribers
        </div>
        <div
          className="srSidebarBio"
        >
          <p>
            
          </p>
        </div>
        <button
          className="srSidebarSubscribeButton"
          onClick={[Function]}
        >
          SUBSCRIBE
        </button>
        <button
          className="srSidebarSubscribeButton"
          onClick={[Function]}
        >
          CREATE A POST
        </button>
      </div>
      <div />
      <div />
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          MODERATORS & ADMINS
        </h5>
        <ul />
      </div>
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          RULES
        </h5>
        <ol />
      </div>
      <div />
      <div />
      <div
        className="subredditSidebarComponent flairComp"
      >
        <h5>
          CREATE YOUR OWN NAME
        </h5>
        <form
          id="flairForm"
          onSubmit={[Function]}
        >
          <input
            className="flairInput"
            placeholder="Flair"
            type="text"
          />
          <button
            className="srSidebarSubscribeButton"
            type="submit"
          >
            Create a flair
          </button>
        </form>
      </div>
    </aside>
  </div>
</div>
`;

exports[`Subreddit component it matches the snapshot 1`] = `
<div
  className="subredditFixed"
>
  <section
    id="subredditShowcase"
    style={
      Object {
        "backgroundImage": null,
      }
    }
  >
    <img
      alt="Subreddit Default"
      src="subreddit.png"
    />
    <h1>
      r/
      
    </h1>
  </section>
  <nav
    id="subredditNavbar"
  >
    <div
      className="subredditContainer"
    >
      <div
        id="subredditSort"
      >
        <div
          className="subredditContainer"
        >
          <div />
        </div>
      </div>
    </div>
  </nav>
  <div
    class="subredditContainer"
  >
    <section
      id="subredditPageContent"
    >
      <div />
    </section>
    <aside
      id="subredditSidebarContainer"
    >
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          COMMUNITY DETAILS
        </h5>
        <div
          className="srSidebarCompTitle"
        >
          <img
            alt="Subreddit Default"
            src="subreddit.png"
          />
          r/
          
        </div>
        <div
          className="srSidebarSubscribers"
        >
          0
           
          <br />
          Subscribers
        </div>
        <div
          className="srSidebarBio"
        >
          <p>
            
          </p>
        </div>
        <button
          className="srSidebarSubscribeButton"
          onClick={[Function]}
        >
          SUBSCRIBE
        </button>
        <button
          className="srSidebarSubscribeButton"
          onClick={[Function]}
        >
          CREATE A POST
        </button>
      </div>
      <div />
      <div />
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          MODERATORS & ADMINS
        </h5>
        <ul />
      </div>
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          RULES
        </h5>
        <ol />
      </div>
      <div />
      <div />
      <div
        className="subredditSidebarComponent flairComp"
      >
        <h5>
          CREATE YOUR OWN NAME
        </h5>
        <form
          id="flairForm"
          onSubmit={[Function]}
        >
          <input
            className="flairInput"
            placeholder="Flair"
            type="text"
          />
          <button
            className="srSidebarSubscribeButton"
            type="submit"
          >
            Create a flair
          </button>
        </form>
      </div>
    </aside>
  </div>
</div>
`;

exports[`Subreddit component snapshot it matches the snapshot 1`] = `
<div
  className="subredditFixed"
>
  <section
    id="subredditShowcase"
    style={
      Object {
        "backgroundImage": null,
      }
    }
  >
    <img
      alt="Subreddit Default"
      src="subreddit.png"
    />
    <h1>
      r/
      
    </h1>
  </section>
  <nav
    id="subredditNavbar"
  >
    <div
      className="subredditContainer"
    >
      <div
        id="subredditSort"
      >
        <div
          className="subredditContainer"
        >
          <div />
        </div>
      </div>
    </div>
  </nav>
  <div
    class="subredditContainer"
  >
    <section
      id="subredditPageContent"
    >
      <div />
    </section>
    <aside
      id="subredditSidebarContainer"
    >
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          COMMUNITY DETAILS
        </h5>
        <div
          className="srSidebarCompTitle"
        >
          <img
            alt="Subreddit Default"
            src="subreddit.png"
          />
          r/
          
        </div>
        <div
          className="srSidebarSubscribers"
        >
          0
           
          <br />
          Subscribers
        </div>
        <div
          className="srSidebarBio"
        >
          <p>
            
          </p>
        </div>
        <button
          className="srSidebarSubscribeButton"
          onClick={[Function]}
        >
          SUBSCRIBE
        </button>
        <button
          className="srSidebarSubscribeButton"
          onClick={[Function]}
        >
          CREATE A POST
        </button>
      </div>
      <div />
      <div />
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          MODERATORS & ADMINS
        </h5>
        <ul />
      </div>
      <div
        className="subredditSidebarComponent"
      >
        <h5>
          RULES
        </h5>
        <ol />
      </div>
      <div />
      <div />
      <div
        className="subredditSidebarComponent flairComp"
      >
        <h5>
          CREATE YOUR OWN NAME
        </h5>
        <form
          id="flairForm"
          onSubmit={[Function]}
        >
          <input
            className="flairInput"
            placeholder="Flair"
            type="text"
          />
          <button
            className="srSidebarSubscribeButton"
            type="submit"
          >
            Create a flair
          </button>
        </form>
      </div>
    </aside>
  </div>
</div>
`;
